### PR TITLE
Websocket Integration

### PR DESCRIPTION
**Pull request by eyingxuan**

### CODE REVIEW

File: frontend/hooks/data-fetching/course.ts [hunk 1]
```diff
@@ -1,4 +1,11 @@
+import { useEffect } from "react";
 import useSWR, { mutate as globalMutate } from "swr";
+import {
+    useRealtimeResourceList,
+    useRealtimeResource,
+} from "@pennlabs/rest-live-hooks";
+// TODO: REMOVE THIS AS SOON AS WE REFACTOR
```
Comment (eyingxuan) [inline frontend/hooks/data-fetching/course.ts:7]: Just wanted to point this out, I think this is the best solution for now - I don't want to spend time refactoring out everything and delaying websockets, but their types don't coerce well so I have to import this version

File: frontend/components/Course/StudentQueuePage/QuestionForm.tsx [hunk 2]
```diff
@@ -1,14 +1,10 @@
 import React, { useState } from "react";
 import { Segment, Form, Header, Button } from "semantic-ui-react";
+import { mutateResourceListFunction } from "@pennlabs/rest-hooks/dist/types";
```
Comment (eyingxuan) [inline frontend/components/Course/StudentQueuePage/QuestionForm.tsx:3]: @davish Should we export types from the library? Or did you choose not to export this for good reason?

Comment (davish) [reply frontend/components/Course/StudentQueuePage/QuestionForm.tsx:3]: yeah, we should. I wish there was a way for us to alias this to `rest-hooks/types` but I'll put it in the index exports in the next release (wow that sounds so professional)

**Comment by eyingxuan** [comment]
Things to polish
- [ ] Avoid subscriptions when queue is closed (Comment: I'm not sure how to do this as of now - you can't wrap a hook in a conditional, and I'm not sure what effect passing undefined to the realtime hook will affect the logic)
- [x] Add orderby for queue resource list
- [ ] Figure out proxy https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/router.md
- [x] Prefix all routes with API instead of using global provider so the POST/PATCH/PUTS will play nice

**Comment by eyingxuan** [comment]
Leaving some notes here for upstream ~frontend-core bugs~ I agree that this should not be the default behavior
- [x] An UPDATED action might have to trigger a revalidate (updating a question status might cause it to disappear from the view): this should probably be an option

**Review by ArmaanT — APPROVED** [review]
This approval was made under duress

**Pull request merged**